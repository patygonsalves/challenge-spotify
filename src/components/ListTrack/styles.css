
.listTrackX-container {
  align-items: center;
  flex-direction: row;
  display: flex;
  margin-bottom: 12px;
}

.listTrackX-text {
  font-size: 14px;
  color: #999999;
}

.listTrackX-description {
  font-size: 16px;
  color: #fafafa;
  margin-left: 20px;
  min-width: 420px;
}