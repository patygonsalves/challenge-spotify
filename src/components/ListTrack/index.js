import React from 'react'
import PropTypes from 'prop-types'

import './styles.css'

const ListTrack = ({
  number,
  name,
  time,
}) => (
  <li className='listTrackX-container'>
    <span className='listTrackX-text'>{number}.</span>
    <span className='listTrackX-description'>{name}</span>
    <span className='listTrackX-text'>{time}</span>
  </li>
)

ListTrack.propTypes = {
  number: PropTypes.number,
  name: PropTypes.string,
  time: PropTypes.string,
}

export default ListTrack
