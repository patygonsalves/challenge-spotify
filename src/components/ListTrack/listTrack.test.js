import React from 'react'
import '@testing-library/jest-dom/extend-expect'

import { render, screen } from '@testing-library/react'

import ListTrack from '.'

describe('<ListTrack />', () => {
  test('should render SearchField', () => {
    const { container } = render(
      <ListTrack
        number={1}
        name={'folklore'}
        time={'3:50'}
      />
    )

    expect(screen.getByText(/folklore/i)).toBeInTheDocument()
    expect(screen.getByText(/1/i)).toBeInTheDocument()
    expect(screen.getByText(/3:50/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

})
