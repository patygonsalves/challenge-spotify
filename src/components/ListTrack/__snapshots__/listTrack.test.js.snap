// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListTrack /> should render SearchField 1`] = `
<li
  class="listTrackX-container"
>
  <span
    class="listTrackX-text"
  >
    1
    .
  </span>
  <span
    class="listTrackX-description"
  >
    folklore
  </span>
  <span
    class="listTrackX-text"
  >
    3:50
  </span>
</li>
`;
